src/TestExplorer/TestRunner.ts

```diff
@@ -233,138 +184,129 @@ export class TestRunner {
+                },
+            });
+
+            if (token.isCancellationRequested) {
+                parsedOutputStream.end();
+                outputStream.end();
+                return;
             }
-            // output test logging to debug console so we can catch it with a tracker
-            testBuildConfig.terminal = "console";
-            return testBuildConfig;
+
+            await this.launchTests(
+                testKind,
+                token,
+                parsedOutputStream,
+                outputStream,
+                testBuildConfig,
+                runState
+            );
         }
     }
 
-    /** Run test session without attaching to a debugger */
-    async runSession(
-        token: vscode.CancellationToken,
+    private async launchTests(
         testKind: TestKind,
+        token: vscode.CancellationToken,
+        stdout: stream.Writable,
+        stderr: stream.Writable,
+        testBuildConfig: vscode.DebugConfiguration,
         runState: TestRunnerTestRunState
     ) {
-        // create launch config for testing
-        const testBuildConfig = this.createLaunchConfigurationForTesting(false);
-        if (testBuildConfig === null) {
-            return;
-        }
-        const testRegex = this.testRegex;
-        // Parse output from stream and output to log
-        const parsedOutputStream = new stream.Writable({
-            write: (chunk, encoding, next) => {
-                const text = chunk.toString();
-                this.testRun.appendOutput(text.replace(/\n/g, "\r\n"));
-                this.testOutputParser.parseResult(text, runState, testRegex);
-                next();
-            },
-        });
-
-        // Output test from stream
-        const outputStream = new stream.Writable({
-            write: (chunk, encoding, next) => {
-                const text = chunk.toString();
-                this.testRun.appendOutput(text.replace(/\n/g, "\r\n"));
-                next();
-            },
-        });
-
-        if (token.isCancellationRequested) {
-            parsedOutputStream.end();
-            outputStream.end();
-            return;
-        }
-
         this.testRun.appendOutput(`> Test run started at ${new Date().toLocaleString()} <\r\n\r\n`);
         try {
             switch (testKind) {
                 case TestKind.coverage:
-                    await this.runCoverageSession(
-                        token,
-                        parsedOutputStream,
-                        outputStream,
-                        testBuildConfig
-                    );
+                    await this.runCoverageSession(token, stdout, stderr, testBuildConfig);
                     break;
                 case TestKind.parallel:
-                    await this.runParallelSession(
-                        token,
-                        parsedOutputStream,
-                        outputStream,
-                        testBuildConfig
-                    );
```

Comment:
Both parallel and coverage sessions don't seem to run swift testing tests

Reply:
This will come in a subsequent patch as there are some changes in SwiftPM required to support our coverage workflow. Specifically we need the `--enable-code-coverage` flag to be added to `swift build` (see https://github.com/apple/swift-package-manager/pull/7518). 

I have the code written, I just need to test the SwiftPM update with a nightly toolchain, which I should be able to do shortly.